src/Conversion/ONNXToKrnl/Tensor/OneHot.cpp

```diff
@@ -0,0 +1,189 @@
+//===---------------- OneHot.cpp - Lowering OneHot Op -------------------===//
+//
+// This file lowers the ONNX OneHot Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp"
+#include "src/Dialect/ONNX/ONNXShapeHelper.hpp"
+
+using namespace mlir;
+
+struct ONNXOneHotOpLowering : public ConversionPattern {
+  ONNXOneHotOpLowering(MLIRContext *ctx)
+      : ConversionPattern(mlir::ONNXOneHotOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    ONNXOneHotOpAdaptor operandAdaptor(operands);
+    ONNXOneHotOp oneHotOp = llvm::cast<ONNXOneHotOp>(op);
+    auto loc = op->getLoc();
+
+    Value indices = operandAdaptor.indices();
+    Value depth = operandAdaptor.depth();
+    Value values = operandAdaptor.values();
+
+    auto depthShape = depth.getType().cast<MemRefType>().getShape();
+    auto indicesShape = indices.getType().cast<MemRefType>().getShape();
+    int64_t indicesRank = indicesShape.size();
+
+    // Axis is a required attribute and should have default value of -1.
```

Comment:
More precisely, axis is an attribute with default value. You are right to use axis() to get the integer value, instead of using aixsAttr() to get the attribute.

Reply:
You may change the comment "required attribute". Not a big deal.